From 80a55e490422af8db1f40a941ca1a12ffc35e117 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Fri, 14 Sep 2018 22:57:36 +0200
Subject: [PATCH 1/1] efi_loader: check parameter new_handle of efi_file_open()

We should not check parameter file twice.
We should check parameter new_handle.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 fs/fat/fat_write.c        | 2 ++
 lib/efi_loader/efi_file.c | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/fs/fat/fat_write.c b/fs/fat/fat_write.c
index fc211e74bc..eb14018560 100644
--- a/fs/fat/fat_write.c
+++ b/fs/fat/fat_write.c
@@ -5,6 +5,8 @@
  * R/W (V)FAT 12/16/32 filesystem implementation by Donggeun Kim
  */
 
+#define DEBUG 1
+
 #include <common.h>
 #include <command.h>
 #include <config.h>
diff --git a/lib/efi_loader/efi_file.c b/lib/efi_loader/efi_file.c
index ef34cc6cd7..40ae7e4b38 100644
--- a/lib/efi_loader/efi_file.c
+++ b/lib/efi_loader/efi_file.c
@@ -218,7 +218,7 @@ static efi_status_t EFIAPI efi_file_open(struct efi_file_handle *file,
 		  open_mode, attributes);
 
 	/* Check parameters */
-	if (!file || !file || !file_name) {
+	if (!file || !new_handle || !file_name) {
 		ret = EFI_INVALID_PARAMETER;
 		goto out;
 	}
-- 
2.18.0

